kong/clustering/rpc/manager.lua

```diff
@@ -141,6 +142,120 @@ function _M:_find_node_and_check_capability(node_id, cap)
 end
 
 
+-- CP => DP
+function _M:_handle_meta_call(c, node_id)
+  local data, typ, err = c:recv_frame()
+  if err then
+    return nil, err
+  end
+
+  assert(typ == "binary")
```

Comment:
Is it proper to handle errors with assert? This seems to cause 500 responses (but in this case it should be a client error, 40x)

Reply:
We did the same thing in socket.lua

Reply:
@chronolaw They are in a callback, which will then be caught; here it's different

Reply:
done